### DISCUSSION TITLE
Suggestion for inlay-hints

### DISCUSSION THREAD

**Discussion by nahojer** [helix-editor/helix#6597] [General]
First of all, I love the inlay hints. Great work! 

**I have a suggestion to make the inlay hints configurable with regards to what kind of inlay hints should be shown.** I apoligize in advance if this is already on the roadmap, but I couldn't find a similar issue that address this.

Consider the following Go snippet that makes a HTTP POST request.

![image](https://user-images.githubusercontent.com/88558001/229691299-65ee5117-10c0-4161-901d-1faf82157213.png)

The inlay hints for the struct field initialization (line 34) and method parameters (e.g. line 39) are great. However, in my opinion, the inlay hints for the variable assignments (e.g. line 29 and 44) is making the code harder to read as I already know the return types -- it feels like stuttering. Below snippet hopefully makes this even clearer.

![image](https://user-images.githubusercontent.com/88558001/229691939-58fad5ef-1fc6-492c-8224-4f93a0f89ad4.png)

**Comment by pascalkuthe** [comment]
Inlay hints are provided by the language server and not helix itself (as everything language specific, helix does not and will.never comtain language specific code). You should raise this suggestion with gopls. There might even already be a setting for that so check the documentation first

**Reply by nahojer** [reply]
Aha, I didn't know this. Makes a lot of sense. Thanks for clarifying!
